lint-rules.md

```diff
@@ -0,0 +1,1281 @@
+        Int align_windows_per_read_n_max = 10000
+        Int align_transcripts_per_window_n_max = 100
+        Int align_transcripts_per_read_n_max = 10000
+        Int pe_overlap_n_bases_min = 0
+        Int win_anchor_multimap_n_max = 50
+        Int win_bin_n_bits = 16
+        Int win_anchor_dist_n_bins = 9
+        Int win_flank_n_bins = 4
+        Int chim_segment_min = 0
+        Int chim_score_min = 0
+        Int chim_score_drop_max = 20
+        Int chim_score_separation = 10
+        Int chim_score_junction_non_GTAG = -1
+        Int chim_junction_overhang_min = 20
+        Int chim_segment_read_gap_max = 0
+        Int chim_main_degment_mult_n_max = 10
+        Int chim_multimap_n_max = 0
+        Int chim_multimap_score_range = 1
+        Int chim_non_chim_score_drop_min = 20
+        Int twopass1_reads_n = -1
+        Int ncpu = 8
+        Int modify_disk_size_gb = 0
+    }
+```
+
+### `disallowed_input_name`
+
+Any input name matching these regular expressions will be flagged: `/^[iI]n[A-Z_]/` or `/^input/i`. It is redundant and needlessly verbose to use an input's name to specify that it is an input. Input names should be short yet descriptive. Prefixing a name with `in` or `input` adds length to the name without adding clarity or context.
+
+**Group**: `naming`
+
+#### Example
+
+All the below input names would be flagged:
+
+```wdl
+    input {
+        File inBam  # This name will be flagged twice, once for the 'in' prefix, and once for being camalCase when it should be snake_case
+        File in_bam
+        File input_gtf
+    }
+```
+
+### `disallowed_output_name`
+
+Any output name matching these regular expressions will be flagged: `/^[oO]ut[A-Z_]/`, `/^output/i` or `/^..?$/`. It is redundant and needlessly verbose to use an output's name to specify that it is an output. Output names should be short yet descriptive. Prefixing a name with `out` or `output` adds length to the name without adding clarity or context. Additionally, names with only 2 characters can lead to confusion and obfuscates the content of an output. Output names should be at least 3 characters long.
+
+**Group**: `naming`
+
+#### Example
+
+All the below input names would be flagged:
+
+```wdl
+    input {
+        File outBam  # This name will be flagged twice, once for the 'out' prefix, and once for being camalCase when it should be snake_case
+        File out_bam
+        File output_gtf
+        String af
+    }
+```
+
+### `no_curly_commands`
+
+`command` blocks should be wrapped with arrows (`<<<` `>>>`) instead of curly brackets (`{` `}`). Certain Bash constructions cause problems with parsing the bracket notation. There are no such problems with the arrow notation.
+
+#### Example
+
+Good:
+
+```wdl
+    command <<<
+        echo "Hello, World!"
+    >>>
+```
+
+Bad:
+
+```wdl
+    command {
+        echo "Hello, World!"
+    }
+```
+
+### `inpersistent_container`  && `persistent_container_not_tagged`
```

Comment:
First of all, if we keep this rule, I would use the word "immutable" instead of "persistent".